drivers/sx127x/Kconfig

```diff
@@ -25,33 +25,22 @@ menuconfig MODULE_SX127X
     help
         Only LoRa long range modem is supported at the moment.
 
-choice SX127X_VARIANT
-    bool "Radio variant"
-    depends on MODULE_SX127X
-    default MODULE_SX1272 if HAVE_SX1272
-    default MODULE_SX1276 if HAVE_SX1276
-
 config MODULE_SX1272
     bool "SX1272"
+    select MODULE_SX127X
```

Comment:
The usage of choices for the variant was a design decision for all drivers. We know that it requires to select both the driver module and variant, but that is only needed when the variant is to be forced. In case the board has the device connected to it, the `HAVE_*` symbol would be selected, indicating a preference for the choice. 

We have considered the approach you have here in the past (actually we started modelling like that), but we realized that it would not allow us to use modules that enable defaults (`MODULE_NETDEV_DEFAULT` in this case). Now, when such default is enabled, the driver module would be enabled (if the correct feature is present) and the correct variant would be the default choice option. This can't be done when the entry-point is the choice. There is some info in this PR https://github.com/RIOT-OS/RIOT/pull/17556/files#diff-2132175f34702f4d76287f4df5c5f5db02d097c2778e5ac47da4c88ef8f13acdR517.

Reply:
From the board with hardwired radio perspective that could make sense. From the end user that plugged the radio on its board, I think it doesn't make sense since he won't have the `HAVE_` symbol defined for his board (thinking of a nucleo64 where one plugs an mbed lora shield), resulting in the user having to enable both sx127x and sx1276 (or sx1272 depending on the shield he is using).

Offline @fjmolinas told me that the choice was here to prevent both sx1272 and sx1276 to be enabled at the same time. But in make one could do that already. I would rather stick to the same behavior as make as a first step and after think of solutions for this problem later.